nifi-extension-bundles/nifi-network-bundle/nifi-network-processors/src/main/java/org/apache/nifi/processors/network/SplitPCAP.java

```diff
@@ -0,0 +1,224 @@
+
+        List<Packet> newPackets = new ArrayList<>();
+        templatePcap.packets().clear();
+
+
+        // Loop through all packets in the pcap file and split them into smaller pcap files.
+        while (!unprocessedPackets.isEmpty()){
+            Packet packet = unprocessedPackets.getFirst();
+
+            if (packet.inclLen() > pcapMaxSize){
+                session.putAttribute(flowFile, ERROR_REASON, "PCAP contains a packet larger than the max size.");
+                session.transfer(flowFile, REL_FAILURE);
+                return;
+            }
+
+            if (currentPacketCollectionSize + (packet.inclLen() + packetHeaderLength) > pcapMaxSize && currentPacketCollectionSize > 0) {
+                templatePcap.packets().addAll(newPackets);
+                var newFlowFile = session.create(flowFile);
+
+                session.write(newFlowFile, out -> out.write(templatePcap.readBytesFull()));
+
+                session.putAttribute(
+                    newFlowFile,
+                    "filename",
+                    flowFile.getAttribute("filename").split("\\.")[0] + "-" + totalFlowfileCount + ".pcap"
+                );
+
+                splitFilesList.add(newFlowFile);
+
+                totalFlowfileCount += 1;
+
+                newPackets = new ArrayList<>();
+                currentPacketCollectionSize = PCAPHeaderLength;
+                templatePcap.packets().clear();
+
+            } else {
+                newPackets.add(packet);
+                currentPacketCollectionSize += ((int) packet.inclLen() + packetHeaderLength);
+                unprocessedPackets.removeFirst();
+            }
+        }
+
+        // If there are any packets left over, create a new flowfile.
+        if(!newPackets.isEmpty()){
+            templatePcap.packets().addAll(newPackets);
+            var newFlowFile = session.create(flowFile);
+            session.putAttribute(
+                newFlowFile,
+                "filename",
+                flowFile.getAttribute("filename").split("\\.")[0] + "-" + totalFlowfileCount + ".pcap"
+            );
+
+            session.write(newFlowFile, out -> out.write(templatePcap.readBytesFull()));
+            splitFilesList.add(newFlowFile);
+        }
+
+        final String fragmentId = UUID.randomUUID().toString();
+
+        int fragmentIndex = 0;
+        final String originalFileName = flowFile.getAttribute("filename");
+
+        for(FlowFile split : splitFilesList){
+            session.putAttribute(split, FRAGMENT_COUNT, String.valueOf(splitFilesList.size()));
+            session.putAttribute(split, FRAGMENT_ID, fragmentId);
+            session.putAttribute(split, FRAGMENT_INDEX, Integer.toString(fragmentIndex));
+            session.putAttribute(split, SEGMENT_ORIGINAL_FILENAME, originalFileName);
+            fragmentIndex++;
+            session.transfer(split, REL_SPLIT);
+        }
+
+        session.transfer(flowFile, REL_ORIGINAL);
+    }
```

Comment:
A few things to mention here. 

1. Per Mark Payne towards the end of the following [thread](https://lists.apache.org/thread/7zo2px31r3377c7vhby4h6nrngdf3llf) mentioned "you should avoid using session.putAttribute" many times. Hence it would be better to place all the attributes in a java.util.Map and then use putAllAttributes. I will demonstrate below.
2. Not necessary to transfer each flow file individually. `session.transfer` is overloaded to take a `java.util.Collection` of FlowFile objects. I will demonstrate below.
3. Add the filename attribute but use `org.apache.nifi.flowfile.attributes.CoreAttributes.FILENAME.key()` for accessing the old filename and inserting the value for each split file name.
4. The parsing of filename assumes there is only a period right before the extension.  The logic should be refactored to account for possible other periods

```suggestion
         final String fragmentId = UUID.randomUUID().toString();
        final String originalFileName = flowFile.getAttribute(CoreAttributes.FILENAME.key());
        final String originalFileNameWithoutExtension = originalFileName.substring(0, originalFileName.lastIndexOf("."));
        
        IntStream.range(0, splitFilesList.size()).forEach(index -> {
            FlowFile split = splitFilesList.get(index);
            Map<String, String> attributes = new HashMap<>();
            attributes.put(CoreAttributes.FILENAME.key(), originalFileNameWithoutExtension + "-" + index + ".pcap");
            attributes.put(FRAGMENT_COUNT, String.valueOf(splitFilesList.size()));
            attributes.put(FRAGMENT_ID, fragmentId);
            attributes.put(FRAGMENT_INDEX, Integer.toString(index));
            attributes.put(SEGMENT_ORIGINAL_FILENAME, originalFileName);
            session.putAllAttributes(split, attributes);
        });

        session.transfer(splitFilesList);
        session.transfer(flowFile, REL_ORIGINAL);
```

Reply:
@JackHintonSmartDCSIT I just noticed Github flagged my recent comments as outdated. I assume that is because I submitted them after your last commit so the files I commented on are no longer in the same package as when I commented on them.